src/libPMacc/PMaccConfig.cmake

```diff
@@ -207,6 +279,8 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_NO_VARIADIC_TEMPLATES")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_NO_CXX11_VARIADIC_TEMPLATES")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_NO_FENV_H")
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    add_definitions(-DBOOST_NO_CXX11_SMART_PTR)
```

Comment:
The error without the suppression is:
```
In file included from /bigdata/hplsim/scratch/widera/dev/src/picongpu/main.cu:33:
In file included from /bigdata/hplsim/scratch/widera/dev/src/picongpu/include/simulation_defines.hpp:
53:
In file included from /bigdata/hplsim/scratch/widera/testPic/khi/include/simulation_defines/unitless/
starter.unitless:25:
In file included from /bigdata/hplsim/scratch/widera/dev/src/picongpu/include/plugins/PluginControlle
r.hpp:58:
In file included from /bigdata/hplsim/scratch/widera/dev/src/picongpu/include/plugins/output/images/P
ngCreator.hpp:33:
In file included from /home/widera/lib/boost_1_62_0_clang4/include/boost/thread.hpp:23:
In file included from /home/widera/lib/boost_1_62_0_clang4/include/boost/thread/barrier.hpp:21:
/home/widera/lib/boost_1_62_0_clang4/include/boost/thread/detail/nullary_function.hpp:9
9:29: error: no member named 'impl' in 'boost::detail::thread_move_t<boost
::detail::nullary_function<void ()> >'; did you mean to use '->' instead of '.'?

      impl(boost::move(other.impl))
                            ^
/home/widera/lib/boost_1_62_0_clang4/include/boost/thread/detail/nullary_function.hpp:1
18:33: error: no member named 'impl' in 'boost::detail::thread_move_t<boos
t::detail::nullary_function<void ()> >'; did you mean to use '->' instead of '.'?

        impl = boost::move(other.impl);
                                ^
/home/widera/lib/boost_1_62_0_clang4/include/boost/thread/detail/nullary_function.hpp:1
98:30: error: no member named 'impl' in 'boost::detail::thread_move_t<boos
t::detail::nullary_function<unsigned long ()> >'
      impl(boost::move(other.impl))
                       ~~~~~ ^
/home/widera/lib/boost_1_62_0_clang4/include/boost/thread/barrier.hpp:203:7: 
note: in instantiation of member function 'boost::detail::nullary_function<unsigned long
 ()>::nullary_function' requested here
      fct_(funct
      ^
3 errors generated.
```